pkg/util/lock_pool.go

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package util
+
+import (
+	"reflect"
+	"sync"
+)
+
+// LockPool is a synchronized set of maps that can be obtained with a provided
+// ID.
+type LockPool[K any, V sync.Locker] struct {
```

Comment:
Does this pool need to be maintained outside of cache? the key is the same, and it looks the lifecycle of each lock pool entry needs to be in sync with the cache item, i.e. when item is cleaned, it needs to be removed from the map here.

Reply:
Good point. I have added a new function `InitOvfCacheAndLockPool` under the `vmprovider.go` file to clean up the lock when the same item expires from the cache.